staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -30,7 +30,7 @@ import (
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/klog/v2"
 
-	"github.com/imdario/mergo"
+	"dario.cat/mergo"
```

Comment:
hrmm, I thought we were dropping use of mergo

Reply:
only for the more problematic case, refreshing your memory of your comment in https://github.com/kubernetes/kubernetes/issues/107499#issuecomment-1938831712 :)

> The trickiest spot we have is loader.go, which relies on one order of merging for shallow struct fields and another order of merging for map fields. This is the spot I'd consider hand-writing and eliminating use of mergo, something like this:

Reply:
ah, too bad... I thought we'd be able to drop it completely, especially since it pulls in gopkg.in/yaml.v3

Reply:
will give it a whirl in a follow up PR @liggitt

Reply:
I'm _really_ unsure our current test coverage would catch bugs and behavior changes in mergo...

for example, I can drop most of the `mergo.WithOverride` options and no test cases fail:
```diff
diff --git a/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go b/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go
index 4f78612c7f7..c1437859f3d 100644
--- a/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go
+++ b/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go
@@ -241,13 +241,13 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		if err != nil {
 			return nil, err
 		}
-		mergo.Merge(clientConfig, userAuthPartialConfig, mergo.WithOverride)
+		mergo.Merge(clientConfig, userAuthPartialConfig)
 
 		serverAuthPartialConfig, err := getServerIdentificationPartialConfig(configAuthInfo, configClusterInfo)
 		if err != nil {
 			return nil, err
 		}
-		mergo.Merge(clientConfig, serverAuthPartialConfig, mergo.WithOverride)
+		mergo.Merge(clientConfig, serverAuthPartialConfig)
 	}
 
 	return clientConfig, nil
@@ -268,7 +268,7 @@ func getServerIdentificationPartialConfig(configAuthInfo clientcmdapi.AuthInfo,
 	configClientConfig.CAData = configClusterInfo.CertificateAuthorityData
 	configClientConfig.Insecure = configClusterInfo.InsecureSkipTLSVerify
 	configClientConfig.ServerName = configClusterInfo.TLSServerName
-	mergo.Merge(mergedConfig, configClientConfig, mergo.WithOverride)
+	mergo.Merge(mergedConfig, configClientConfig)
 
 	return mergedConfig, nil
 }
@@ -338,8 +338,8 @@ func (config *DirectClientConfig) getUserIdentificationPartialConfig(configAuthI
 		promptedConfig := makeUserIdentificationConfig(*promptedAuthInfo)
 		previouslyMergedConfig := mergedConfig
 		mergedConfig = &restclient.Config{}
-		mergo.Merge(mergedConfig, promptedConfig, mergo.WithOverride)
-		mergo.Merge(mergedConfig, previouslyMergedConfig, mergo.WithOverride)
+		mergo.Merge(mergedConfig, promptedConfig)
+		mergo.Merge(mergedConfig, previouslyMergedConfig)
 		config.promptedCredentials.username = mergedConfig.Username
 		config.promptedCredentials.password = mergedConfig.Password
 	}
@@ -507,7 +507,7 @@ func (config *DirectClientConfig) getContext() (clientcmdapi.Context, error) {
 
 	mergedContext := clientcmdapi.NewContext()
 	if configContext, exists := contexts[contextName]; exists {
-		mergo.Merge(mergedContext, configContext, mergo.WithOverride)
+		mergo.Merge(mergedContext, configContext)
 	} else if required {
 		return clientcmdapi.Context{}, fmt.Errorf("context %q does not exist", contextName)
 	}
@@ -525,12 +525,12 @@ func (config *DirectClientConfig) getAuthInfo() (clientcmdapi.AuthInfo, error) {
 
 	mergedAuthInfo := clientcmdapi.NewAuthInfo()
 	if configAuthInfo, exists := authInfos[authInfoName]; exists {
-		mergo.Merge(mergedAuthInfo, configAuthInfo, mergo.WithOverride)
+		mergo.Merge(mergedAuthInfo, configAuthInfo)
 	} else if required {
 		return clientcmdapi.AuthInfo{}, fmt.Errorf("auth info %q does not exist", authInfoName)
 	}
 	if config.overrides != nil {
-		mergo.Merge(mergedAuthInfo, config.overrides.AuthInfo, mergo.WithOverride)
+		mergo.Merge(mergedAuthInfo, config.overrides.AuthInfo)
 	}
 
 	return *mergedAuthInfo, nil
@@ -543,7 +543,7 @@ func (config *DirectClientConfig) getCluster() (clientcmdapi.Cluster, error) {
 
 	mergedClusterInfo := clientcmdapi.NewCluster()
 	if config.overrides != nil {
-		mergo.Merge(mergedClusterInfo, config.overrides.ClusterDefaults, mergo.WithOverride)
+		mergo.Merge(mergedClusterInfo, config.overrides.ClusterDefaults)
 	}
 	if configClusterInfo, exists := clusterInfos[clusterInfoName]; exists {
 		mergo.Merge(mergedClusterInfo, configClusterInfo, mergo.WithOverride)
```

The same change on master also passes tests, but I cannot tell if that's because our test coverage is weak for these call sites (either not exercising them at all or not exercising override merge cases), or if the WithOverride option is irrelevant.